### PR TITLE
contentChanged was never thrown, because the text from Monaco Editor …

### PR DESCRIPTION
**Pull request by spec2e**
…was not seen by the context model. Wired the two models together.

Not sure if this is the "correct" way to do this, but it fixed the problem. 
If you do not agree, can you then please tell me how to fix it? 
Thanks :-)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -74,12 +74,28 @@ class MonacoWidget extends Widget implements DocumentRegistry.IReadyWidget {
...
     this.editor = monaco.editor.create(this.node, {
-      model: monaco.editor.createModel(content, undefined, uri)
+      model: monaco_model
     });
+
+    let the_editor = this.editor;
+    let the_model = this.context.model;
+
+    monaco_model.onDidChangeContent((event) => {
+      the_model.value.text = the_editor.getValue();
```
Comment (jasongrout) [inline src/index.ts]: Because we are using an arrow function, you can directly use `this`.

Comment (spec2e) [reply src/index.ts]: OK - I'll fix it

**Comment by gnestor** [comment]
I tested this and it seems to fix the issue of documents not saving 👍

**Comment by haobibo** [comment]
👍Yes, before this PR, modifications in the Monaco editor won't be saved.
Here is an example to test this PR:
https://gitlab.com/labq/labq/blob/master/docker_jpy/Dockerfile

**Comment by jasongrout** [comment]
Thanks!

**Pull request merged**